### DISCUSSION TITLE
TS0601 TZE200_yojqa8xn gas sensor not supported

### DISCUSSION THREAD

**Discussion by sercio** [Koenkk/zigbee2mqtt#14492] [Devices]
Hello,
i have this device, I think is similar to TS0601_gas_sensor. This is the log.

info  2022-07-18 09:59:59: Device '0xa4c1381aea7ee7a6' joined
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381aea7ee7a6","ieee_address":"0xa4c1381aea7ee7a6"},"type":"device_joined"}'
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0xa4c1381aea7ee7a6"},"type":"device_connected"}'
info  2022-07-18 09:59:59: Starting interview of '0xa4c1381aea7ee7a6'
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381aea7ee7a6","ieee_address":"0xa4c1381aea7ee7a6","status":"started"},"type":"device_interview"}'
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0xa4c1381aea7ee7a6"},"type":"pairing"}'
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381aea7ee7a6","ieee_address":"0xa4c1381aea7ee7a6"},"type":"device_announce"}'
info  2022-07-18 09:59:59: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0xa4c1381aea7ee7a6"},"type":"device_announced"}'
info  2022-07-18 10:00:00: Successfully interviewed '0xa4c1381aea7ee7a6', device has successfully been paired
warn  2022-07-18 10:00:00: Device '0xa4c1381aea7ee7a6' with Zigbee model 'TS0601' and manufacturer name '_TZE200_yojqa8xn' is NOT supported, please follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
info  2022-07-18 10:00:00: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":null,"friendly_name":"0xa4c1381aea7ee7a6","ieee_address":"0xa4c1381aea7ee7a6","status":"successful","supported":false},"type":"device_interview"}'
info  2022-07-18 10:00:00: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"friendly_name":"0xa4c1381aea7ee7a6","supported":false},"type":"pairing"}'
warn  2022-07-18 10:00:01: Received message from unsupported device with Zigbee model 'TS0601' and manufacturer name '_TZE200_yojqa8xn'
warn  2022-07-18 10:00:01: Please see: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

Anyone have the same problem?
Thanks

**Comment by irekzielinski** [comment]
Have the same device.
Tried by creating custom JS file:

`
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require("zigbee-herdsman-converters/lib/tuya");

const definition = {
    // Since a lot of Tuya devices use the same modelID, but use different data points
    // it's usually necessary to provide a fingerprint instead of a zigbeeModel
    fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0601' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS0601',
            // The manufacturer name from: Device with modelID 'TS0601' is not supported.
            manufacturerName: '_TZE200_yojqa8xn'
        },
    ],
    model: 'TS0601_gas_sensor', // Vendor model number, look on the device for a model number
    vendor: 'TuYa', // Vendor of the device (only used for documentation and startup logging)
    description: 'Gas Sensor', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [
        fz.ignore_basic_report, // Add this if you are getting no converter for 'genBasic'
        fz.tuya_data_point_dump, // This is a debug converter, it will be described in the next part
        fz.tuya_gas
    ],
    toZigbee: [
        tz.tuya_data_point_test, // Another debug converter
    ],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
    },
    exposes: [
        // Here you should put all functionality that your device exposes
        e.gas()
    ],
};

module.exports = definition;
`

But don't get correct gas readings. Only this messsage in the log:
zigbee-herdsman-converters:tuya_data_point_dump: Received Tuya DataPoint #2 from 0xa4c138a07437d162 with raw data '{"dp":2,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,1]}}': type='commandDataReport', datatype='value', value='1', known DP# usage: ["heatingSetpoint","coverPosition","eardaDimmerLevel","moesHold","moesSheatingSetpoint","silvercrestChangeMode","tuyaSabCO2","tuyaSahkMP25","tuyaSabCO","moes105DimmerLevel1","trsSensitivity","trsfSensitivity","tvMode","haozeeHeatingSetpoint","nousHumidity","tthHumidity","tIlluminanceLux","evanellMode","AM02PercentControl","x5hMode","connecteMode","tshpscSensitivity","alectoSmokeValue"]

Not sure how to progress that

**Reply by Koenkk** [reply]
Can you figure out the data points? https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#requirements-and-caveats

**Reply by irekzielinski** [reply]
sadly I don't have Tuya Gateway :(

**Reply by sercio** [reply]
> sadly I don't have Tuya Gateway :(

There is a issue at this link:
https://github.com/Koenkk/zigbee2mqtt/issues/14571

Can you test? For me it's impossible now.

**Reply by sercio** [reply]
Hello,
today I have received the tuya zigbee gateway.... the datapoints are more different then the converter ... I have found:

1 Gas detection state
2 Detected Gas
6 Alarm Ringtone
7 Alarm Time
8 Auto-detect
9 Auto-detect Result
10 Preheat

I don't understand how to find the datatype and the values for create a correct converter... can anyone help me?
